### DISCUSSION TITLE
Catalyst 21.11

### DISCUSSION THREAD

**Discussion by Scitator** [catalyst-team/catalyst#1366] [Announcements]
### Tl;dr
Framework architecture simplification and speedup + SSL & RecSys extensions.

### Added

- MultiVAE RecSys example ([#1340](https://github.com/catalyst-team/catalyst/pull/1340))`
- Returned `resume` support - resolved [#1193](https://github.com/catalyst-team/catalyst/issues/1193) ([#1349](https://github.com/catalyst-team/catalyst/pull/1349))
- Smoothing dice loss to contrib ([#1344](https://github.com/catalyst-team/catalyst/pull/1344))
- `profile` flag for `runner.train` ([#1348](https://github.com/catalyst-team/catalyst/pull/1348))
- MultiDAE RecSys example ([#1356](https://github.com/catalyst-team/catalyst/pull/1356))
- `SETTINGS.log_batch_metrics`, `SETTINGS.log_epoch_metrics`, `SETTINGS.compute_per_class_metrics` for framework-wise  Metric & Logger APIs specification ([#1357](https://github.com/catalyst-team/catalyst/pull/1357))
- `log_batch_metrics` and `log_epoch_metrics` options for all available Loggers ([#1357](https://github.com/catalyst-team/catalyst/pull/1357))
- `compute_per_class_metrics` option for all available multiclass/label metrics ([#1357](https://github.com/catalyst-team/catalyst/pull/1357))
- pytorch benchmark script and simplified MNIST ([#1360](https://github.com/catalyst-team/catalyst/pull/1360))

### Changed

- A few framework simplifications were made ([#1346](https://github.com/catalyst-team/catalyst/pull/1346)):
  - `catalyst-contrib` scripts reduced to `collect-env` and `project-embeddings` only
  - `catalyst-dl` scripts recuded to `run` and `tune` only
  - `transforms.` prefix deprecated for Catalyst-based transforms
  - `catalyst.tools` moved to `catalyst.extras`
  - task-dependent extensions from `catalyst.data` moved to `catalyst.contrib.data`
  - `catalyst.data.transforms` moved to `catalyst.contrib.data.transforms`
  - `Normalize`, `ToTensor` transforms renamed to `NormalizeImage`, `ImageToTensor`
  - metric learning extensions moved to `catalyst.contrib.data`
  - `catalyst.contrib` moved to code-as-a-documentation development
  - `catalyst[cv]` and `catalyst[ml]` extensions moved to flatten architecture design; examples: `catalyst.contrib.data.dataset_cv`, `catalyst.contrib.data.dataset_ml`
  - `catalyst.contrib` moved to flatten architecture design; exampels: `catalyst.contrib.data`, `catalyst.contrib.datasets`, `catalyst.contrib.layers`, `catalyst.contrib.models`, `catalyst.contrib.optimizers`, `catalyst.contrib.schedulers`
  - internal functionality moved to `***._misc` modules
  - `catalyst.utils.mixup` moved to `catalyst.utils.torch`
  - `catalyst.utils.numpy` moved to `catalyst.contrib.utils.numpy`
- default logging logic moved from "batch & epoch" to "epoch"-only to save computation time during logging; to respecify, please use:
  - `SETTINGS.log_batch_metrics=True/False` or `os.environ["CATALYST_LOG_BATCH_METRICS"]`
  - `SETTINGS.log_epoch_metrics=True/False` or `os.environ["CATALYST_LOG_EPOCH_METRICS"]`
- default metrics computation moved from "per-class & aggregations" to "aggregations"-only to save computation time during logging; to respecify, please use:
  - `SETTINGS.compute_per_class_metrics=True/False` or `os.environ["CATALYST_COMPUTE_PER_CLASS_METRICS"]`
- no transformations required for MNIST contrib dataset ([#1360](https://github.com/catalyst-team/catalyst/pull/1360)

### Removed

- A few framework simplifications were made ([#1346](https://github.com/catalyst-team/catalyst/pull/1346)):
  - `catalyst.contrib.pandas`
  - `catalyst.contrib.parallel`
  - `catalyst.contrib.models.cv`
  - a few `catalyst.utils.misc` functions
  - `catalyst.extras` removed from the public documentation

### Fixed

- documentation search error (21.10 only) ([#1346](https://github.com/catalyst-team/catalyst/pull/1346))
- docs examples ([#1362](https://github.com/catalyst-team/catalyst/pull/1362))
- Self-Supervised benchmark: ([#1365](https://github.com/catalyst-team/catalyst/pull/1365)), ([#1361](https://github.com/catalyst-team/catalyst/pull/1361))

### Contributors ❤️
@asteyo @Dokholyan @Nimrais @y-ksenia @sergunya17 

<hr /><em>This discussion was created from the release <a href='https://github.com/catalyst-team/catalyst/releases/tag/v21.11'>Catalyst 21.11</a>.</em>
